compiler/rustc_data_structures/src/sync.rs

```diff
@@ -598,6 +782,10 @@ pub struct OneThread<T> {
 unsafe impl<T> std::marker::Sync for OneThread<T> {}
 #[cfg(parallel_compiler)]
 unsafe impl<T> std::marker::Send for OneThread<T> {}
+#[cfg(parallel_compiler)]
+unsafe impl<T> DynSync for OneThread<T> {}
+#[cfg(parallel_compiler)]
+unsafe impl<T> DynSend for OneThread<T> {}
```

Comment:
Aren't these implied by the impls above?

Reply:
It seems that I still need to `impl DynSend/DynSync` manually, but we can use `assert_send` and `assert_sync` to ensure correctness